### ISSUE TITLE
Multi-Label Image Classification

### ISSUE THREAD

**Issue by sailfish009** [mlbio-epfl/turtle#6]
Hi, I think this is a really good unsupervised learning model, my question is that I know the current implementation is for single label with 1 label for 1 image, is it possible to extend it and apply it to a multilabel classification problem with more than 1 label for 1 image? Off the top of my head, I would have to apply the sigmoid function instead of the argmax function. 

DSEG660: Multi-Label Image Classification 
https://www.kaggle.com/c/hbku2019/data

**Comment by agadetsky** [comment]
Dear @sailfish009,

Since we are tackling fully unsupervised learning/clustering problem, it is unclear what it Multi-Label Clustering problem. Furthermore, in the link provided it seems that your dataset contains ground truth labels, so you are tackling **the supervised learning problem**.

Nevertheless, indeed, one particular dataset can have many valid meaningful clusterings, however the current formulation of our method cannot generate all the valid clustering out of the box. One possible workaround might be to employ different representation spaces reflecting each corresponding valid labeling of your dataset, and run TURTLE independently with each representation space to generate multiple valid labelings of your dataset. For example, imagine that you have a dataset of dogs and cats located indoor and outdoor, and you would like to find both clusterings, i.e., (i) cat vs dog; (ii) indoor vs outdoor. You can try to find/pretrain representations that are well suited for the corresponding desired clustering (one representation space that reflects cat vs dog and the other one that reflects indoor vs outdoor) and run the TURTLE with each of the spaces independently.

Let me know if that solves your issue.

Best,
Artyom

**Comment by sailfish009** [comment]
For now, we need to prepare a valid dataset, so my idea is to use it to apply unsupervised learning to multilabel image classification, since the situation is the same: the training dataset is unlabeled.

The left side of the diagram below is a schematic of the current Turtle, and the right side is a schematic of my idea. 

The idea is not completely organized, but the basic idea is to reverse the order of the process and use the labels and embeddings from the valid dataset to search for similarity to the embeddings in the training data.  

![fun drawio](https://github.com/user-attachments/assets/f8bdbdec-483a-47ee-a5a7-0e126e3b7813)
